File: examples/cs/subscriber.cs
 
using System;
using icehms;
using System.Threading;


class Subscriber : Holon
{
    public Subscriber(IceApp app, string name) : base(app, name)
    {
        log("Starting Holon");
        app.subscribeEvent(this, "MyTopic");
    }

    public override void putMessage(hms.Message msg, Ice.Current current)
    {
        log("Got a message from publisher: " + msg + msg.arguments["counter"]);
    }
}


class Test {
    static void Main() {
        Console.WriteLine ("Starting");
        IceApp app = null;
        Subscriber sub = null;
        try
        {
            app = new IceApp("MyTestAdapter", "localhost", 12000);
            sub = new Subscriber(app, "MyTestHolon");
            Thread.Sleep(100000);
        }
        catch (Exception ex)
        {
            Console.WriteLine("Catching" + ex);
        }
        finally
        {
            if (sub != null )
            {
                sub.shutdown();
            }
            if ( app != null ) 
            {
                app.shutdown();
            }
        }
    }
}

File: examples/py/subscriber.py
#!/usr/bin/python
import sys
from struct import unpack
import logging

from icehms import Holon, run_holon, hms

#class Client(Holon, hms.GenericEventInterface):
class Client(Holon):
    def __init__(self, name, logLevel=logging.INFO ):
        Holon.__init__(self, name, logLevel=logLevel )
        self._tn = "MyTopic"

    def run(self):
        #self._subscribeTopic(self._tn)
        self._subscribe(self._tn)

    def put_message(self, msg, current=None):
        print("Got message: ", msg)


if __name__ == "__main__":
    s = Client("MyTopicClient", topicname)
    run_holon(s)
